src/components/Area.js

```diff
@@ -0,0 +1,693 @@
+  start: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  sweep: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+};
+
+Area.Arc.defaultProps = {
+  ...AreaComponentDefaultProps,
+  start: 0,
+};
+
+Area.Circle = class Circle extends AreaComponent {
+  getWidth(p) {
+    return 2 * this.parseParent(this.props.r, p);
+  }
+
+  getHeight(p) {
+    return getWidth(p);
+  }
+
+  draw(area, p) {
+    const path = new libui.UiDrawPath(0 /*uiDrawFillModeWinding*/);
+    path.newFigureWithArc(
+      this.parseParent(this.props.x, p),
+      this.parseParent(this.props.y, p, true),
+      this.parseParent(this.props.r, p),
+      0,
+      2 * Math.PI,
+      false
+    );
+    path.end();
+    return path;
+  }
+};
+
+Area.Circle.PropTypes = {
+  ...AreaComponentPropTypes,
+  x: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+  y: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+  r: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+};
+
+Area.Circle.defaultProps = {
+  ...AreaComponentDefaultProps,
+};
+
+Area.Bezier = class Bezier extends AreaComponent {
+  draw(area, p) {
+    const path = new libui.UiDrawPath(0 /*uiDrawFillModeWinding*/);
+    path.newFigure(
+      this.parseParent(this.props.x1, p),
+      this.parseParent(this.props.y1, p, true)
+    );
+    path.bezierTo(
+      this.parseParent(this.props.cx1, p),
+      this.parseParent(this.props.cy1, p, true),
+      this.parseParent(this.props.cx2, p),
+      this.parseParent(this.props.cy2, p, true),
+      this.parseParent(this.props.x2, p),
+      this.parseParent(this.props.y2, p, true)
+    );
+    path.end();
+
+    return path;
+  }
+};
+
+Area.Bezier.PropTypes = {
+  ...AreaComponentPropTypes,
+  x1: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+  y1: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+  cx1: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+  cy1: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+  x2: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+  y2: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+  cx2: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+  cy2: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+};
+
+Area.Path = class Path extends AreaComponent {
+  draw(area, p) {
+    const path = new libui.UiDrawPath(
+      this.props.fillRule === 'evenodd' ? 1 : 0
+    );
+    const commands = parseSVG(this.props.d);
+    parseSVG.makeAbsolute(commands);
+
+    for (let i = 0; i < commands.length; i++) {
+      const c = commands[i];
+      switch (c.command) {
+        case 'moveto':
+          path.newFigure(c.x, c.y);
+          break;
+
+        case 'quadratic curveto':
+          console.log("Quadratic Beziers aren't implemented!");
```

Comment:
Maybe make this throw instead of output?

Reply:
If you want. Currently, it falls back to a line because of the `switch`:
```js
switch(cmd){
  case 'quadratic curveto':
    // ...
  case 'horizontal lineto':
  case 'vertical lineto':
    path.lineTo(c.x, c.y);
    break;
}
```

Reply:
Should we throw here as well?
```js
switch(){
  // ...
  default:
    console.log('Not implemented in Path:', c);
}
```

Reply:
Yeah it would be better to prevent confusion and unexpected things happening.

Reply:
Done

Reply:
Done:

```
```